contracts/WakuRlnRegistry.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import {WakuRln} from "./WakuRln.sol";
+import {IPoseidonHasher} from "rln-contract/PoseidonHasher.sol";
+import {Ownable} from "openzeppelin-contracts/contracts/access/Ownable.sol";
+import "forge-std/console.sol";
+
+error StorageAlreadyExists(address storageAddress);
+error NoStorageContractAvailable();
+error IncompatibleStorage();
+error IncompatibleStorageIndex();
+
+contract WakuRlnRegistry is Ownable {
+    uint16 public nextStorageIndex;
+    mapping(uint16 => address) public storages;
+
+    uint16 public usingStorageIndex = 0;
+
+    IPoseidonHasher public immutable poseidonHasher;
+
+    event NewStorageContract(uint16 index, address storageAddress);
```

Comment:
```suggestion
    event NewStorageContract(uint16 indexed index, address indexed storageAddress);
```

Reply:
good idea! however, I am not sure if this is too useful in the `indexed` context - why would someone need to fetch the `NewStorageContract` with a specific contractIndex / storageAddress?
With erc20 transfers, its useful to filter based on the `_from` address, since that gives us all the transactions with the `_from` address. These events will be only one-time, hence using `indexed` may not be the best. curious about your thoughts though